plugins/woocommerce/includes/abstracts/abstract-wc-order.php

```diff
@@ -103,6 +103,14 @@ abstract class WC_Abstract_Order extends WC_Abstract_Legacy_Order {
 	 */
 	protected $object_type = 'order';
 
+	/**
+	 * Indicates if set_parent_id will throw an error if no order exists with the supplied id.
+	 *
+	 * @since 7.9.0
+	 * @var bool
+	 */
+	private $verify_parent_id = true;
```

Comment:
Seems like this is residual logic that was removed in #38199 , if so then let's remove it.

Reply:
Implemented in https://github.com/woocommerce/woocommerce/pull/37050/commits/5e1624be4662215f6fbf98d6bb6bccaff56f2a5d